includes/classes/Customer.php

```diff
@@ -645,6 +645,30 @@ public function getNumberOfOrders(): int
         return $result->fields['total'];
     }
 
+    /**
+     * Return a simple summary of orders for this customer, ordered most recent first.
+     *
+     * @param integer $limit Optional limit of the number of orders to return.
+     * @return array
+     */
+    public function getOrdersSummary(int $limit = 0): array {
```

Comment:
Given that `getOrderHistory()` already exists, I wonder if this is needed. Sure, that other function checks more tables and brings back more data, so might be slightly slower. 

Perhaps if this function is kept, naming it `getRecentOrderRecords()` might be slightly clearer about its intent?

Reply:
Tiny point on formatting: while we used to keep the opening `{` for a function/method on the same line in years past, in recent years we're putting that opening brace on a new line when it's for a function declaration or class declaration. (We still keep it inline for `for`, `foreach`, `if` etc).

Reply:
You're right, I was scared off using `getOrderHistory()` because it looked like too much was going on, and it constructed a `splitPageResult`.  Taking a new look, I have optimised the query so the product count is done within the initial query, but I ran into trouble with the `splitPageResult` constructor in `admin` being different to the store front, i.e. https://github.com/zencart/zencart/blob/1da3aa3efc5fbdce79333c06b10bc738c223e230/admin/includes/classes/split_page_results.php#L92 vs https://github.com/zencart/zencart/blob/1da3aa3efc5fbdce79333c06b10bc738c223e230/includes/classes/split_page_results.php#L53

What's the deal with this code drift?  We cannot use all `Customer` methods from admin because of it.

On the formatting, that's cool.  I was afraid to run `cs-fixer` since it would probably create hundreds of modifications :)